R/modify.R

```diff
@@ -174,7 +174,7 @@ imodify <- function(.x, .f, ...) {
 modify_where <- function(.x, .where, .f, ..., .error_call = caller_env()) {
   if (vec_is_list(.x)) {
     out <- vec_proxy(.x)
-    out[.where] <- map(out[.where], .f, ...)
+    out[.where] <- no_zap(map(out[.where], .f, ...), .error_call)
     vec_restore(out, .x)
   } else if (is.data.frame(.x)) {
```

Comment:
Do you want to use `no_zap()` in this branch too for consistent errors? Right now the snapshot is telling me it looks like it errors in `vec_size()`, which is fine, at least it errors, but this might be clearer?